### PR TITLE
Init access the data

### PR DESCRIPTION
**Pull request by nyarly**
This is as embarrassingly huge PR. Sorry for that.

Contained within are the additions and changes I needed to make in order to get Access the Data ready to deploy. Note well! AtD isn't successfully deployed yet, but that has to do with issues with CKAN, not the incubator platform.

# Introduction
(yeah, it's that kind of PR)

As a rough guide to what's changed:
The headline is the `multi-container-service` module. It's designed with AtD in mind, but in general I suspect that being able to deploy e.g. a web backend and Redis (say) will be really valuable to Hack for LA. MCS also covers details like setting up public and private DNS. MCS covers a couple other outstanding Incubator issues, c.f. #18

MCS also drives an `ecr` module, to create container registries that developers can push to to update their services.

Other new modules include `database` which uses the `cyrilgdn/postgresql` provider to provision postgres databases and roles within the shared RDS instance. All credentials are stored in SSM parameters as `SecureString`s, so there's no toxic secrets to manage in IaC. (c.f. #1)

Because that's such a useful pattern, there's also a `cheap-secrets` module, which allows for purpose-agnostic secrets to be securely generated and stashed in SSM. This isn't 100% ideal, but the benefit is that managing secrets this way is free, as compared to using AWS SecretStore. The downside is that the secret material might be stored in terraform state files and would theoretically be extractable. The same personnel who could do that, though, could simply query SSM, or deploy a task that would forward secrets to them.

The Terraform `shared_resources` modules have been fleshed out significantly, such that we can get data from all of them.

Overall, I feel very good about these changes, and would recommend that future Incubator projects be built using `multi-container-service` on AtD's pattern _even if_ they only deploy a single container.

### CODE REVIEW

File: terraform-incubator/access-the-data/versions.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Get configuration from the shared infrastructure
```
Comment (nyarly) [inline terraform-incubator/access-the-data/versions.tf:1]: Projects wanting to copy from AtD should be able to copy this file verbatim. There's some boilerplate that should be the same for every project.

File: terraform-incubator/access-the-data/main.tf [hunk 2]
```diff
@@ -0,0 +1,148 @@
+locals {
```
Comment (nyarly) [inline terraform-incubator/access-the-data/main.tf:1]: The contents of this file were designed to contain exactly what varies between projects. Other projects might crib from this, but expect to change basically every value.

File: terraform-incubator/shared_resources/all/main.tf [hunk 3]
```diff
@@ -26,31 +26,72 @@ data "terraform_remote_state" "shared" {
   }
 }
 
+locals {
+  configuration = {
```
Comment (nyarly) [inline terraform-incubator/shared_resources/all/main.tf:30]: Prefer always to use the single shared `configuration` object - it reduces the coupling between modules, who only need to name the fields that they care about.

Each field is exported as an `output` separately, to support legacy use.

File: terraform-modules/database/main.tf [hunk 4]
```diff
@@ -0,0 +1,89 @@
+terraform {
```
Comment (nyarly) [inline terraform-modules/database/main.tf:1]: Ideally, we would migrate to this to the complete exclusion of the DB lambda

File: terraform-modules/ecs-task/main.tf [hunk 5]
```diff
@@ -0,0 +1,208 @@
+locals {
```
Comment (nyarly) [inline terraform-modules/ecs-task/main.tf:1]: This was copied from the existing `ecs` module and modified for use in MCS.

File: terraform-modules/multi-container-service/tasks.tf [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  desired_count    = each.value.desired_count
+  container_cpu    = each.value.cpu
+  container_memory = each.value.memory
+
+  host_names        = [for s in each.value.subdomains : "${s}.${data.aws_route53_zone.this.name}"]
+  path_patterns     = each.value.path_patterns
+  health_check_path = each.value.health_check_path
+  log_group         = aws_cloudwatch_log_group.cwlogs.name
+  container_port    = each.value.port
+  container_env_vars = merge(each.value.env_vars, {
```
Comment (chelseybeck) [inline terraform-modules/multi-container-service/tasks.tf:90]: @nyarly I see a few formatting things like this w/ alignment...maybe run a final tf fmt?

**Review by chelseybeck — APPROVED** [review]
Great work, @nyarly!

**Pull request merged**